app/src/main/java/android/support/v17/leanback/supportleanbackshowcase/app/media/VideoMediaPlayerGlue.java

```diff
@@ -91,16 +112,89 @@ public boolean abandonAudioFocus() {
         return mAudioManager.abandonAudioFocus(this) == AudioManager.AUDIOFOCUS_REQUEST_GRANTED;
     }
 
+    private void updateVideoSessionMetaData() {
+        if (mMediaMetaData == null) {
+            throw new IllegalArgumentException(
+                    "mCurrentMediaItem is null in updateMediaSessionMetaData!");
+        }
+        final MediaMetadata.Builder metaDataBuilder = new MediaMetadata.Builder();
+        if (mMediaMetaData.getMediaTitle() != null) {
+            metaDataBuilder.putString(MediaMetadata.METADATA_KEY_TITLE,
+                    mMediaMetaData.getMediaTitle());
+        }
+        if (mMediaMetaData.getMediaAlbumName() != null) {
+            metaDataBuilder.putString(MediaMetadata.METADATA_KEY_ALBUM,
+                    mMediaMetaData.getMediaAlbumName());
+        }
+        if (mMediaMetaData.getMediaArtistName() != null) {
+            metaDataBuilder.putString(MediaMetadata.METADATA_KEY_ARTIST,
+                   mMediaMetaData.getMediaArtistName());
+        }
+        Resources res = getContext().getResources();
+        int cardWidth = res.getDimensionPixelSize(R.dimen.playback_now_playing_card_width);
+        int cardHeight = res.getDimensionPixelSize(R.dimen.playback_now_playing_card_height);
+        if (mMediaMetaData.getMediaAlbumArtUrl() != null) {
+            Glide.with(getContext())
+                    .load(mMediaMetaData.getMediaAlbumArtUrl())
+                    .asBitmap()
+                    .into(new SimpleTarget<Bitmap>(cardWidth, cardHeight) {
+                              @Override
+                              public void onResourceReady(
+                                      Bitmap resource,
+                                      GlideAnimation<? super Bitmap> glideAnimation) {
+                                  metaDataBuilder.putBitmap(MediaMetadata.METADATA_KEY_ALBUM_ART,
```

Comment:
I would be surprised if Glide load bitmap in main thread.  This seems creating a glide task telling it to post to SimpleTarget on main thread after bitmap is loaded in background thread.